src/generator/AutoRest.Ruby.Azure/TemplateModels/AzureMethodTemplateModel.cs

```diff
@@ -97,13 +106,14 @@ public string InvokeNextMethodAsync()
                         builder.AppendLine(string.Format(CultureInfo.InvariantCulture, "{0} = {1}", param.Name, argumentName));
                     }
                 }
+                builder.Outdent();
             }
 
-            IList<string> headerParams = nextMethod.Parameters.Where(p => (p.Location == ParameterLocation.Header || p.Location == ParameterLocation.None) && !p.IsConstant && p.ClientProperty == null).Select(p => p.Name).ToList();
-            headerParams.Add("custom_headers");
```

Comment:
Nope but we don't need that explicit addition as `nextMethodTemplateModel.MethodParameterInvocation` includes it as shown in below example

```
    def get_multiple_pages_fragment_next_link_as_lazy(api_version, tenant, custom_headers = nil)
      response = get_multiple_pages_fragment_next_link_async(api_version, tenant, custom_headers).value!
      unless response.nil?
        page = response.body
        page.next_method = Proc.new do |next_link|
          next_fragment_async(api_version, tenant, next_link, custom_headers)
        end
        page
      end
    end
```